src/App/Fossa/Analyze.hs

```diff
@@ -86,20 +91,38 @@ data ScanDestination
 -- | UnpackArchives bool flag
 data UnpackArchives = UnpackArchives
 
-analyzeMain :: BaseDir -> Bool -> Severity -> ScanDestination -> OverrideProject -> Flag UnpackArchives -> [BuildTargetFilter] -> IO ()
-analyzeMain basedir debugMode logSeverity destination project unpackArchives filters =
+-- | "Replay logging" modes
+data RecordMode =
+    RecordModeRecord -- ^ record effect invocations
+  | RecordModeReplay FilePath -- ^ replay effect invocations from a file
+  | RecordModeNone -- ^ don't record or replay
+
+analyzeMain :: FilePath -> RecordMode -> Severity -> ScanDestination -> OverrideProject -> Flag UnpackArchives -> [BuildTargetFilter] -> IO ()
+analyzeMain workdir recordMode logSeverity destination project unpackArchives filters =
   withLogger logSeverity
     . Diag.logWithExit_
     . runReadFSIO
     . runExecIO
-    $ if not debugMode
-      then analyze basedir destination project unpackArchives filters
-      else do
+    $ case recordMode of
+      RecordModeNone -> do
```

Comment:
Really glad to see this not tied to debug mode.  I was going to file this as a separate issue until I saw it here.